scripts/Engineering/gui/engineering_diffraction/tabs/calibration/model.py

```diff
@@ -146,144 +154,167 @@ def update_calibration_params_table(params_table):
-        curve_plot_bank_1.plot(van_curves_ws, wkspIndex=1)
-        curve_plot_bank_1.plot(van_curves_ws, wkspIndex=2)
-        curve_plot_bank_1.set_title("Engg GUI Vanadium Curves Bank 1")
-        curve_plot_bank_1.legend(["Data", "Calc", "Diff"])
-
-        curve_plot_bank_2.plot(van_curves_ws, wkspIndex=3)
-        curve_plot_bank_2.plot(van_curves_ws, wkspIndex=4)
-        curve_plot_bank_2.plot(van_curves_ws, wkspIndex=5)
-        curve_plot_bank_2.set_title("Engg GUI Vanadium Curves Bank 2")
-        curve_plot_bank_2.legend(["Data", "Calc", "Diff"])
-
-        fig.show()
-
-    @staticmethod
-    def _generate_tof_fit_workspace(difa, difc, tzero, bank):
-        bank_ws = Ads.retrieve(CalibrationModel._generate_table_workspace_name(bank))
-
-        x_val = []
-        y_val = []
-        y2_val = []
-
-        difa_to_plot = difa
-        difc_to_plot = difc
-        tzero_to_plot = tzero
-
-        for irow in range(0, bank_ws.rowCount()):
-            x_val.append(bank_ws.cell(irow, 0))
-            y_val.append(bank_ws.cell(irow, 5))
-            y2_val.append(pow(x_val[irow], 2) * difa_to_plot + x_val[irow] * difc_to_plot + tzero_to_plot)
-
-        ws1 = CreateWorkspace(DataX=x_val,
-                              DataY=y_val,
-                              UnitX="Expected Peaks Centre (dSpacing A)",
-                              YUnitLabel="Fitted Peaks Centre(TOF, us)")
-        ws2 = CreateWorkspace(DataX=x_val, DataY=y2_val)
-
-        output_ws = "engggui_tof_peaks_bank_" + str(bank)
-        if Ads.doesExist(output_ws):
-            DeleteWorkspace(output_ws)
-
-        AppendSpectra(ws1, ws2, OutputWorkspace=output_ws)
-        DeleteWorkspace(ws1)
-        DeleteWorkspace(ws2)
-
     def _plot_tof_fit(self):
         bank_1_ws = Ads.retrieve("engggui_tof_peaks_bank_1")
         bank_2_ws = Ads.retrieve("engggui_tof_peaks_bank_2")
+        bank_1_residuals_ws = Ads.retrieve("engggui_tof_peaks_bank_1_residuals")
+        bank_2_residuals_ws = Ads.retrieve("engggui_tof_peaks_bank_2_residuals")
         # Create plot
         fig = plt.figure()
-        gs = gridspec.GridSpec(1, 2)
-        plot_bank_1 = fig.add_subplot(gs[0], projection="mantid")
-        plot_bank_2 = fig.add_subplot(gs[1], projection="mantid")
+        gs = gridspec.GridSpec(2, 2)
+        plot_bank_1 = fig.add_subplot(gs[0, 0], projection="mantid")
+        plot_bank_2 = fig.add_subplot(gs[0, 1], projection="mantid")
+        plot_res_1 = fig.add_subplot(gs[1, 0], projection="mantid")
+        plot_res_2 = fig.add_subplot(gs[1, 1], projection="mantid")
 
         for ax, ws, bank in zip([plot_bank_1, plot_bank_2], [bank_1_ws, bank_2_ws], [1, 2]):
             self._add_plot_to_axes(ax, ws, bank)
+        for ax, ws in zip([plot_res_1, plot_res_2], [bank_1_residuals_ws, bank_2_residuals_ws]):
+            self._add_residuals_to_axes(ax, ws)
+        fig.tight_layout()
         fig.show()
 
     def _plot_tof_fit_single_bank_or_custom(self, bank):
```

Comment:
Is there not a way of combining this function with `_plot_tof_fit`? Once you have established how many banks (i.e. how many columns of subplots) should this not just be a loop over nbanks (which may be 1)?